src/main/java/service/impl/BalanceCalculatorImpl.java

```diff
@@ -0,0 +1,38 @@
+package service.impl;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import model.FruitTransaction;
+import service.OperationHandler;
+import service.OperationService;
+import strategy.OperationStrategy;
+
+public class BalanceCalculatorImpl implements OperationHandler {
+    @Override
+    public Map<String, Integer> update(List<FruitTransaction> data) {
+        Set<String> fruitsSet = new HashSet<>();
+        for (FruitTransaction fruit : data) {
+            fruitsSet.add(fruit.getFruit());
+        }
+        Map<String, Integer> fruitsMap = new HashMap<>();
```

Comment:
Storage represents real DB. So create a Storage class with a static public Map and use it instead of local variable